### PR TITLE
Enables eslint on the full repo and adds a rule for no `only()` tests

### PR DESCRIPTION
**Pull request by tony-sull**
## Changes

- Enables eslint on all packages except webapi (requires too many linter fixes for now)
- Adds a lint rule to error on `only()` tests
- Fixes all linter errors in the repo (mostly `no-console` and `no-shadow` errors)

[Example CI failure](https://github.com/withastro/astro/pull/3659/checks?check_run_id=6987518161) when a pull request had a `.only()` test 🎉 

## Testing

N/A

## Docs

N/A

### CODE REVIEW

File: .eslintignore [hunk 1]
```diff
@@ -1,8 +1,9 @@
-**/*.js
-**/*.ts
-!packages/astro/**/*.js
-!packages/astro/**/*.ts
-packages/astro/test/**/*.js
+**/*.d.ts
+packages/**/dist/**/*
+packages/**/fixtures/**/*
```
Comment (tony-sull) [inline .eslintignore:3]: enable all packages, but skip dist and test fixtures

File: .eslintignore [hunk 2]
```diff
@@ -1,8 +1,9 @@
...
+packages/webapi/**/*
```
Comment (tony-sull) [inline .eslintignore:4]: webapi would need a much larger change to fix ~40 linter errors

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -54,9 +54,6 @@ jobs:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
         with:
           eslint: true
-          eslint_args: --ignore-pattern test --ignore-pattern vendor
```
Comment (tony-sull) [inline .github/workflows/ci.yml:57]: Allowing eslint in CI to use the same .eslintignore and .eslintrc.js config files

File: packages/astro/test/tailwindcss.test.js [hunk 4]
```diff
@@ -62,10 +60,10 @@ describe('Tailwind', () => {
 
 		it('handles Markdown pages', async () => {
 			const html = await fixture.readFile('/markdown-page/index.html');
-			const $ = cheerio.load(html);
-			const bundledCSSHREF = $('link[rel=stylesheet][href^=/assets/]').attr('href');
-			const bundledCSS = await fixture.readFile(bundledCSSHREF.replace(/^\/?/, '/'));
-			expect(bundledCSS, 'includes used component classes').to.match(/\.bg-purple-600{/);
+			const $md = cheerio.load(html);
```
Comment (tony-sull) [inline packages/astro/test/tailwindcss.test.js:63]: Was shadowing the `$` variable shared by all other tests that were loading `/index.html`

**Comment by matthewp** [comment]
lgtm

**Review by natemoo-re — APPROVED** [review]
The `only()` PR I've ever wanted! Approved.

> My apologies to @bholmesdev

**Pull request merged**